src/views/PurchaseOrder.vue

```diff
@@ -119,30 +150,51 @@ export default defineComponent({
           showToast(translate("Something went wrong, Please try again"));
         }
       },
-
       async parseFile(){
         await parseCsv(this.file).then(res => {
           this.content = res;
         })
       },
-      mapFields() {
-        const areAllFieldsSelected = Object.values(this.fields).every(field => field !== "");
+      review() {
+        const areAllFieldsSelected = Object.values(this.fieldMapping).every(field => field !== "");
         if (this.content.length <= 0) {
           showToast(translate("Please upload a valid purchase order csv to continue"));
         } else if (areAllFieldsSelected) {
-          this.orderItemsList = this.content.map(item => ({
-            orderId: item[this.fields.orderId],
-            shopifyProductSKU: item[this.fields.productSku],
-            arrivalDate: item[this.fields.date],
-            quantityOrdered: item[this.fields.quantity],
-            externalFacilityId: item[this.fields.facility]
-          }))
+          this.orderItemsList = this.content.map(item => {
+            return {
+              orderId: item[this.fieldMapping.orderId],
+              shopifyProductSKU: item[this.fieldMapping.productSku],
+              arrivalDate: DateTime.fromFormat(item[this.fieldMapping.orderDate], this.dateTimeFormat).toFormat(this.dateTimeFormat), //This is to verify whether the date format is correct.
+              quantityOrdered: item[this.fieldMapping.quantity],
+              facilityId: '',
+              externalFacilityId: item[this.fieldMapping.facility]
+            }
+          })
           this.store.dispatch('order/updatedOrderList', this.orderItemsList);
           this.router.push({
             name:'PurchaseOrderDetail'
           })
         } else {
           showToast(translate("Select all the fields to continue"));
+        } 
+      },
+      mapFields(event) {
+        if(event && event.detail.value) {
+          const fieldMapping = JSON.parse(JSON.stringify(event.detail.value));
+          const CsvFields = Object.keys(this.content[0]);
+
+          const missingFields = Object.values(fieldMapping.mappingPrefValue).filter(field => {
+            if(!Object.keys(this.content[0]).includes(field)) return field;
+          });
+          if(missingFields.length) showToast(translate(`Some of the mapping fields are missing in the CSV: \n ${ missingFields.join(", ") }`))
```

Comment:
I think we could pass missingFields as variable
Refer this code
https://github.com/hotwax/preorder/blob/main/src/views/product-details.vue#L347